pkg/genericcli/context_test.go

```diff
@@ -0,0 +1,1043 @@
+package genericcli
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"testing"
+	"time"
+
+	"github.com/fatih/color"
+	"github.com/google/go-cmp/cmp"
+	"github.com/metal-stack/metal-lib/pkg/genericcli/printers"
+	"github.com/metal-stack/metal-lib/pkg/pointer"
+	"github.com/metal-stack/metal-lib/pkg/testcommon"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/require"
+	"sigs.k8s.io/yaml"
+)
+
+// ManagerTestCase defines a test case for ContextManager operations
+type ManagerTestCase[T any] struct {
+	Name        string
+	FileContent *contextConfig
+	Setup       func(t *testing.T, manager *ContextManager) error
+	Run         func(t *testing.T, manager *ContextManager) (T, error)
+	wantErr     error
+	want        T
+}
+
+func ctx1() *Context {
+	return &Context{
+		Name:     "ctx1",
+		APIToken: "token1",
+	}
+}
+
+func ctx2() *Context {
+	return &Context{
+		Name:     "ctx2",
+		APIToken: "token2",
+		Provider: "foo",
+	}
+}
+
+func ctx3() *Context {
+	return &Context{
+		Name:           "ctx3",
+		APIURL:         pointer.Pointer("http://foo.bar"),
+		APIToken:       "token3",
+		DefaultProject: "project3",
+		Timeout:        pointer.Pointer(time.Duration(100)),
+		Provider:       "foo",
+	}
+}
+
+func ctxNew() *Context {
+	return &Context{
+		Name:     "ctxNew",
+		APIToken: "tokenNew",
+	}
+}
+
+func ctxList() []*Context {
+	return []*Context{
+		ctx1(),
+		ctx2(),
+		ctx3(),
+	}
+}
+
+func contextsActiveUnsetCurrentUnset() *contextConfig {
```

Comment:
Maybe rename to `contextContextWithActiveUnsetCurrentUnset`? Same for the following fixture functions below